beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestAddonChangeAlignment.java

```diff
@@ -0,0 +1,192 @@
+import java.util.UUID;
+
+import org.joda.time.LocalDate;
+import org.killbill.billing.account.api.Account;
+import org.killbill.billing.api.TestApiListener.NextEvent;
+import org.killbill.billing.catalog.api.BillingActionPolicy;
+import org.killbill.billing.catalog.api.PlanPhaseSpecifier;
+import org.killbill.billing.entitlement.api.DefaultEntitlementSpecifier;
+import org.killbill.billing.entitlement.api.Subscription;
+import org.killbill.billing.platform.api.KillbillConfigSource;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class TestAddonChangeAlignment extends TestIntegrationBase {
+
+    @Override
+    protected KillbillConfigSource getConfigSource(final Map<String, String> extraProperties) {
+        final Map<String, String> allExtraProperties = new HashMap<>(extraProperties);
+        allExtraProperties.put("org.killbill.catalog.uri", "catalogs/testAddonChangeAlignment");
+        return super.getConfigSource(null, allExtraProperties);
+    }
+
+    @Test(groups = "slow")
+    public void testChangeAlignmentStartOfBundleEOTChange() throws Exception {
+        final LocalDate today = new LocalDate(2023, 8, 1);
+        clock.setDay(today);
+
+        final Account account = createAccountWithNonOsgiPaymentMethod(getAccountData(1));
+
+        //create base 2023-08-01
+        final PlanPhaseSpecifier spec = new PlanPhaseSpecifier("basic-monthly");
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.INVOICE);
+        final UUID baseEntId = entitlementApi.createBaseEntitlement(account.getId(), new DefaultEntitlementSpecifier(spec), null, null, null, false, true, Collections.emptyList(), callContext);
+        assertListenerStatus();
+
+        //Base CTD is 2023-08-01 as expected
+        Subscription baseSub = subscriptionApi.getSubscriptionForEntitlementId(baseEntId, false, callContext);
+        Assert.assertEquals(baseSub.getChargedThroughDate(), new LocalDate(2023, 8, 1));
+
+        clock.setDay(new LocalDate(2023, 8, 5));
+
+        //create addon 2023-08-05
+        final PlanPhaseSpecifier addonSpec = new PlanPhaseSpecifier("BasicAO-monthly");
+        busHandler.pushExpectedEvents(NextEvent.CREATE, NextEvent.BLOCK, NextEvent.INVOICE);
+        final UUID addonEntId = entitlementApi.addEntitlement(baseSub.getBundleId(), new DefaultEntitlementSpecifier(addonSpec), null, null, false, Collections.emptyList(), callContext);
+        assertListenerStatus();
+
+        //ADDON CTD is 2023-08-05 as expected
+        Subscription addonSub = subscriptionApi.getSubscriptionForEntitlementId(addonEntId, false, callContext);
+        Assert.assertEquals(addonSub.getChargedThroughDate(), new LocalDate(2023, 8, 05));
+
+        clock.setDay(new LocalDate(2023, 8, 15));
+
+        //change addon with start of bundle alignment on 2023-08-15 - change will happen immediately as subscription is in TRIAL phase
+        busHandler.pushExpectedEvents(NextEvent.CHANGE, NextEvent.INVOICE);
```

Comment:
Could we add an assertion below on the invoices: Interestingly we end up with 2 FIXED items for the same add-on subscriptions with a different start date - so let's assert on those as this is currently our 'normal' behavior.

Reply:
> Could we add an assertion below on the invoices

Done.

> : Interestingly we end up with 2 FIXED items for the same add-on subscriptions with a different start date

This happens only in the test written on top of #1739. It does not happen in the tests written against `master`. So, the duplicate invoice item is a side effect of changes done for #1739.